src/mip/utils/timestamp.hpp

```diff
@@ -0,0 +1,222 @@
+#pragma once
+
+#include <assert.h>
+#include <chrono>
+
+using std::chrono::duration_cast;
+using Nanoseconds = std::chrono::nanoseconds;
+using Microseconds = std::chrono::microseconds;
+using Milliseconds = std::chrono::milliseconds;
+using Seconds = std::chrono::seconds;
+using Minutes = std::chrono::minutes;
+using Hours = std::chrono::hours;
+
+#if _HAS_CXX20
```

Comment:
The ones above the C++ 20 check are (or should be) yeah. The builds ran with them so that looks good. I haven't run into any issues with them but let me know if you see a potential issue there.

Reply:
Yep, so it looks like the reason days, weeks, years aren't supported on Mac and Linux is because they use an overload of std::chrono::period which is only available in C++ 20 and not yet implemented. The others are supported.